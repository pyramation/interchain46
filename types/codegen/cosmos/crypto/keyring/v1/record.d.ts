import { Any, AnyAmino, AnySDKType } from "../../../../google/protobuf/any";
import { BIP44Params, BIP44ParamsAmino, BIP44ParamsSDKType } from "../../hd/v1/hd";
import * as _m0 from "protobufjs/minimal";
/** Record is used for representing a key in the keyring. */
export interface Record {
    /** name represents a name of Record */
    name: string;
    /** pub_key represents a public key in any format */
    pubKey?: Any;
    /** local stores the private key locally. */
    local?: Record_Local;
    /** ledger stores the information about a Ledger key. */
    ledger?: Record_Ledger;
    /** Multi does not store any other information. */
    multi?: Record_Multi;
    /** Offline does not store any other information. */
    offline?: Record_Offline;
}
/** Record is used for representing a key in the keyring. */
export interface RecordAmino {
    /** name represents a name of Record */
    name: string;
    /** pub_key represents a public key in any format */
    pub_key?: AnyAmino;
    /** local stores the private key locally. */
    local?: Record_LocalAmino;
    /** ledger stores the information about a Ledger key. */
    ledger?: Record_LedgerAmino;
    /** Multi does not store any other information. */
    multi?: Record_MultiAmino;
    /** Offline does not store any other information. */
    offline?: Record_OfflineAmino;
}
/** Record is used for representing a key in the keyring. */
export interface RecordSDKType {
    name: string;
    pub_key?: AnySDKType;
    local?: Record_LocalSDKType;
    ledger?: Record_LedgerSDKType;
    multi?: Record_MultiSDKType;
    offline?: Record_OfflineSDKType;
}
/**
 * Item is a keyring item stored in a keyring backend.
 * Local item
 */
export interface Record_Local {
    privKey?: Any;
}
/**
 * Item is a keyring item stored in a keyring backend.
 * Local item
 */
export interface Record_LocalAmino {
    priv_key?: AnyAmino;
}
/**
 * Item is a keyring item stored in a keyring backend.
 * Local item
 */
export interface Record_LocalSDKType {
    priv_key?: AnySDKType;
}
/** Ledger item */
export interface Record_Ledger {
    path?: BIP44Params;
}
/** Ledger item */
export interface Record_LedgerAmino {
    path?: BIP44ParamsAmino;
}
/** Ledger item */
export interface Record_LedgerSDKType {
    path?: BIP44ParamsSDKType;
}
/** Multi item */
export interface Record_Multi {
}
/** Multi item */
export interface Record_MultiAmino {
}
/** Multi item */
export interface Record_MultiSDKType {
}
/** Offline item */
export interface Record_Offline {
}
/** Offline item */
export interface Record_OfflineAmino {
}
/** Offline item */
export interface Record_OfflineSDKType {
}
export declare const Record: {
    encode(message: Record, writer?: _m0.Writer): _m0.Writer;
    decode(input: _m0.Reader | Uint8Array, length?: number): Record;
    fromJSON(object: any): Record;
    toJSON(message: Record): unknown;
    fromPartial(object: Partial<Record>): Record;
    fromAmino(object: RecordAmino): Record;
    toAmino(message: Record): RecordAmino;
};
export declare const Record_Local: {
    encode(message: Record_Local, writer?: _m0.Writer): _m0.Writer;
    decode(input: _m0.Reader | Uint8Array, length?: number): Record_Local;
    fromJSON(object: any): Record_Local;
    toJSON(message: Record_Local): unknown;
    fromPartial(object: Partial<Record_Local>): Record_Local;
    fromAmino(object: Record_LocalAmino): Record_Local;
    toAmino(message: Record_Local): Record_LocalAmino;
};
export declare const Record_Ledger: {
    encode(message: Record_Ledger, writer?: _m0.Writer): _m0.Writer;
    decode(input: _m0.Reader | Uint8Array, length?: number): Record_Ledger;
    fromJSON(object: any): Record_Ledger;
    toJSON(message: Record_Ledger): unknown;
    fromPartial(object: Partial<Record_Ledger>): Record_Ledger;
    fromAmino(object: Record_LedgerAmino): Record_Ledger;
    toAmino(message: Record_Ledger): Record_LedgerAmino;
};
export declare const Record_Multi: {
    encode(_: Record_Multi, writer?: _m0.Writer): _m0.Writer;
    decode(input: _m0.Reader | Uint8Array, length?: number): Record_Multi;
    fromJSON(_: any): Record_Multi;
    toJSON(_: Record_Multi): unknown;
    fromPartial(_: Partial<Record_Multi>): Record_Multi;
    fromAmino(_: Record_MultiAmino): Record_Multi;
    toAmino(_: Record_Multi): Record_MultiAmino;
};
export declare const Record_Offline: {
    encode(_: Record_Offline, writer?: _m0.Writer): _m0.Writer;
    decode(input: _m0.Reader | Uint8Array, length?: number): Record_Offline;
    fromJSON(_: any): Record_Offline;
    toJSON(_: Record_Offline): unknown;
    fromPartial(_: Partial<Record_Offline>): Record_Offline;
    fromAmino(_: Record_OfflineAmino): Record_Offline;
    toAmino(_: Record_Offline): Record_OfflineAmino;
};
