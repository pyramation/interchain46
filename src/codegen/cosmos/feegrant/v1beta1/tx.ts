import { Any, AnyAmino, AnySDKType } from "../../../google/protobuf/any";
import { BasicAllowance, PeriodicAllowance, AllowedMsgAllowance } from "./feegrant";
import * as _m0 from "protobufjs/minimal";
import { isSet, DeepPartial } from "../../../helpers";
/**
 * MsgGrantAllowance adds permission for Grantee to spend up to Allowance
 * of fees from the account of Granter.
 */

export interface MsgGrantAllowance {
  /** granter is the address of the user granting an allowance of their funds. */
  granter: string;
  /** grantee is the address of the user being granted an allowance of another user's funds. */

  grantee: string;
  /** allowance can be any of basic, periodic, allowed fee allowance. */

  allowance?: (BasicAllowance & PeriodicAllowance & AllowedMsgAllowance & Any) | undefined;
}
export interface MsgGrantAllowanceProtoType {
  typeUrl: "/cosmos.feegrant.v1beta1.MsgGrantAllowance";
  value: Uint8Array;
}
/**
 * MsgGrantAllowance adds permission for Grantee to spend up to Allowance
 * of fees from the account of Granter.
 */

export interface MsgGrantAllowanceAmino {
  /** granter is the address of the user granting an allowance of their funds. */
  granter: string;
  /** grantee is the address of the user being granted an allowance of another user's funds. */

  grantee: string;
  /** allowance can be any of basic, periodic, allowed fee allowance. */

  allowance?: AnyAmino;
}
export interface MsgGrantAllowanceAminoType {
  type: "cosmos-sdk/MsgGrantAllowance";
  value: MsgGrantAllowanceAmino;
}
/**
 * MsgGrantAllowance adds permission for Grantee to spend up to Allowance
 * of fees from the account of Granter.
 */

export interface MsgGrantAllowanceSDKType {
  granter: string;
  grantee: string;
  allowance?: AnySDKType;
}
/** MsgGrantAllowanceResponse defines the Msg/GrantAllowanceResponse response type. */

export interface MsgGrantAllowanceResponse {}
export interface MsgGrantAllowanceResponseProtoType {
  typeUrl: "/cosmos.feegrant.v1beta1.MsgGrantAllowanceResponse";
  value: Uint8Array;
}
/** MsgGrantAllowanceResponse defines the Msg/GrantAllowanceResponse response type. */

export interface MsgGrantAllowanceResponseAmino {}
export interface MsgGrantAllowanceResponseAminoType {
  type: "cosmos-sdk/MsgGrantAllowanceResponse";
  value: MsgGrantAllowanceResponseAmino;
}
/** MsgGrantAllowanceResponse defines the Msg/GrantAllowanceResponse response type. */

export interface MsgGrantAllowanceResponseSDKType {}
/** MsgRevokeAllowance removes any existing Allowance from Granter to Grantee. */

export interface MsgRevokeAllowance {
  /** granter is the address of the user granting an allowance of their funds. */
  granter: string;
  /** grantee is the address of the user being granted an allowance of another user's funds. */

  grantee: string;
}
export interface MsgRevokeAllowanceProtoType {
  typeUrl: "/cosmos.feegrant.v1beta1.MsgRevokeAllowance";
  value: Uint8Array;
}
/** MsgRevokeAllowance removes any existing Allowance from Granter to Grantee. */

export interface MsgRevokeAllowanceAmino {
  /** granter is the address of the user granting an allowance of their funds. */
  granter: string;
  /** grantee is the address of the user being granted an allowance of another user's funds. */

  grantee: string;
}
export interface MsgRevokeAllowanceAminoType {
  type: "cosmos-sdk/MsgRevokeAllowance";
  value: MsgRevokeAllowanceAmino;
}
/** MsgRevokeAllowance removes any existing Allowance from Granter to Grantee. */

export interface MsgRevokeAllowanceSDKType {
  granter: string;
  grantee: string;
}
/** MsgRevokeAllowanceResponse defines the Msg/RevokeAllowanceResponse response type. */

export interface MsgRevokeAllowanceResponse {}
export interface MsgRevokeAllowanceResponseProtoType {
  typeUrl: "/cosmos.feegrant.v1beta1.MsgRevokeAllowanceResponse";
  value: Uint8Array;
}
/** MsgRevokeAllowanceResponse defines the Msg/RevokeAllowanceResponse response type. */

export interface MsgRevokeAllowanceResponseAmino {}
export interface MsgRevokeAllowanceResponseAminoType {
  type: "cosmos-sdk/MsgRevokeAllowanceResponse";
  value: MsgRevokeAllowanceResponseAmino;
}
/** MsgRevokeAllowanceResponse defines the Msg/RevokeAllowanceResponse response type. */

export interface MsgRevokeAllowanceResponseSDKType {}

function createBaseMsgGrantAllowance(): MsgGrantAllowance {
  return {
    granter: "",
    grantee: "",
    allowance: undefined
  };
}

export const MsgGrantAllowance = {
  encode(message: MsgGrantAllowance, writer: _m0.Writer = _m0.Writer.create()): _m0.Writer {
    if (message.granter !== "") {
      writer.uint32(10).string(message.granter);
    }

    if (message.grantee !== "") {
      writer.uint32(18).string(message.grantee);
    }

    if (message.allowance !== undefined) {
      Any.encode((message.allowance as Any), writer.uint32(26).fork()).ldelim();
    }

    return writer;
  },

  decode(input: _m0.Reader | Uint8Array, length?: number): MsgGrantAllowance {
    const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    let end = length === undefined ? reader.len : reader.pos + length;
    const message = createBaseMsgGrantAllowance();

    while (reader.pos < end) {
      const tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.granter = reader.string();
          break;

        case 2:
          message.grantee = reader.string();
          break;

        case 3:
          message.allowance = (FeeAllowanceI_InterfaceDecoder(reader) as Any);
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },

  fromJSON(object: any): MsgGrantAllowance {
    return {
      granter: isSet(object.granter) ? String(object.granter) : "",
      grantee: isSet(object.grantee) ? String(object.grantee) : "",
      allowance: isSet(object.allowance) ? Any.fromJSON(object.allowance) : undefined
    };
  },

  toJSON(message: MsgGrantAllowance): unknown {
    const obj: any = {};
    message.granter !== undefined && (obj.granter = message.granter);
    message.grantee !== undefined && (obj.grantee = message.grantee);
    message.allowance !== undefined && (obj.allowance = message.allowance ? Any.toJSON(message.allowance) : undefined);
    return obj;
  },

  fromPartial(object: Partial<MsgGrantAllowance>): MsgGrantAllowance {
    const message = createBaseMsgGrantAllowance();
    message.granter = object.granter ?? "";
    message.grantee = object.grantee ?? "";
    message.allowance = object.allowance !== undefined && object.allowance !== null ? Any.fromPartial(object.allowance) : undefined;
    return message;
  },

  fromAmino(object: MsgGrantAllowanceAmino): MsgGrantAllowance {
    return {
      granter: object.granter,
      grantee: object.grantee,
      allowance: object?.allowance ? FeeAllowanceI_FromAmino(object.allowance) : undefined
    };
  },

  toAmino(message: MsgGrantAllowance): MsgGrantAllowanceAmino {
    const obj: any = {};
    obj.granter = message.granter;
    obj.grantee = message.grantee;
    obj.allowance = message.allowance ? FeeAllowanceI_ToAmino((message.allowance as Any)) : undefined;
    return obj;
  }

};

function createBaseMsgGrantAllowanceResponse(): MsgGrantAllowanceResponse {
  return {};
}

export const MsgGrantAllowanceResponse = {
  encode(_: MsgGrantAllowanceResponse, writer: _m0.Writer = _m0.Writer.create()): _m0.Writer {
    return writer;
  },

  decode(input: _m0.Reader | Uint8Array, length?: number): MsgGrantAllowanceResponse {
    const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    let end = length === undefined ? reader.len : reader.pos + length;
    const message = createBaseMsgGrantAllowanceResponse();

    while (reader.pos < end) {
      const tag = reader.uint32();

      switch (tag >>> 3) {
        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },

  fromJSON(_: any): MsgGrantAllowanceResponse {
    return {};
  },

  toJSON(_: MsgGrantAllowanceResponse): unknown {
    const obj: any = {};
    return obj;
  },

  fromPartial(_: Partial<MsgGrantAllowanceResponse>): MsgGrantAllowanceResponse {
    const message = createBaseMsgGrantAllowanceResponse();
    return message;
  },

  fromAmino(_: MsgGrantAllowanceResponseAmino): MsgGrantAllowanceResponse {
    return {};
  },

  toAmino(_: MsgGrantAllowanceResponse): MsgGrantAllowanceResponseAmino {
    const obj: any = {};
    return obj;
  }

};

function createBaseMsgRevokeAllowance(): MsgRevokeAllowance {
  return {
    granter: "",
    grantee: ""
  };
}

export const MsgRevokeAllowance = {
  encode(message: MsgRevokeAllowance, writer: _m0.Writer = _m0.Writer.create()): _m0.Writer {
    if (message.granter !== "") {
      writer.uint32(10).string(message.granter);
    }

    if (message.grantee !== "") {
      writer.uint32(18).string(message.grantee);
    }

    return writer;
  },

  decode(input: _m0.Reader | Uint8Array, length?: number): MsgRevokeAllowance {
    const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    let end = length === undefined ? reader.len : reader.pos + length;
    const message = createBaseMsgRevokeAllowance();

    while (reader.pos < end) {
      const tag = reader.uint32();

      switch (tag >>> 3) {
        case 1:
          message.granter = reader.string();
          break;

        case 2:
          message.grantee = reader.string();
          break;

        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },

  fromJSON(object: any): MsgRevokeAllowance {
    return {
      granter: isSet(object.granter) ? String(object.granter) : "",
      grantee: isSet(object.grantee) ? String(object.grantee) : ""
    };
  },

  toJSON(message: MsgRevokeAllowance): unknown {
    const obj: any = {};
    message.granter !== undefined && (obj.granter = message.granter);
    message.grantee !== undefined && (obj.grantee = message.grantee);
    return obj;
  },

  fromPartial(object: Partial<MsgRevokeAllowance>): MsgRevokeAllowance {
    const message = createBaseMsgRevokeAllowance();
    message.granter = object.granter ?? "";
    message.grantee = object.grantee ?? "";
    return message;
  },

  fromAmino(object: MsgRevokeAllowanceAmino): MsgRevokeAllowance {
    return {
      granter: object.granter,
      grantee: object.grantee
    };
  },

  toAmino(message: MsgRevokeAllowance): MsgRevokeAllowanceAmino {
    const obj: any = {};
    obj.granter = message.granter;
    obj.grantee = message.grantee;
    return obj;
  }

};

function createBaseMsgRevokeAllowanceResponse(): MsgRevokeAllowanceResponse {
  return {};
}

export const MsgRevokeAllowanceResponse = {
  encode(_: MsgRevokeAllowanceResponse, writer: _m0.Writer = _m0.Writer.create()): _m0.Writer {
    return writer;
  },

  decode(input: _m0.Reader | Uint8Array, length?: number): MsgRevokeAllowanceResponse {
    const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
    let end = length === undefined ? reader.len : reader.pos + length;
    const message = createBaseMsgRevokeAllowanceResponse();

    while (reader.pos < end) {
      const tag = reader.uint32();

      switch (tag >>> 3) {
        default:
          reader.skipType(tag & 7);
          break;
      }
    }

    return message;
  },

  fromJSON(_: any): MsgRevokeAllowanceResponse {
    return {};
  },

  toJSON(_: MsgRevokeAllowanceResponse): unknown {
    const obj: any = {};
    return obj;
  },

  fromPartial(_: Partial<MsgRevokeAllowanceResponse>): MsgRevokeAllowanceResponse {
    const message = createBaseMsgRevokeAllowanceResponse();
    return message;
  },

  fromAmino(_: MsgRevokeAllowanceResponseAmino): MsgRevokeAllowanceResponse {
    return {};
  },

  toAmino(_: MsgRevokeAllowanceResponse): MsgRevokeAllowanceResponseAmino {
    const obj: any = {};
    return obj;
  }

};
export const FeeAllowanceI_InterfaceDecoder = (input: _m0.Reader | Uint8Array): BasicAllowance | PeriodicAllowance | AllowedMsgAllowance | Any => {
  const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
  const data = Any.decode(reader, reader.uint32());

  switch (data.typeUrl) {
    case "/cosmos.feegrant.v1beta1.BasicAllowance":
      return BasicAllowance.decode(data.value);

    case "/cosmos.feegrant.v1beta1.PeriodicAllowance":
      return PeriodicAllowance.decode(data.value);

    case "/cosmos.feegrant.v1beta1.AllowedMsgAllowance":
      return AllowedMsgAllowance.decode(data.value);

    default:
      return data;
  }
};
export const FeeAllowanceI_FromAmino = (content: AnyAmino) => {
  switch (content.type) {
    case "cosmos-sdk/BasicAllowance":
      return Any.fromPartial({
        typeUrl: "/cosmos.feegrant.v1beta1.BasicAllowance",
        value: BasicAllowance.encode(BasicAllowance.fromPartial((content.value as DeepPartial<BasicAllowance>))).finish()
      });

    case "cosmos-sdk/PeriodicAllowance":
      return Any.fromPartial({
        typeUrl: "/cosmos.feegrant.v1beta1.PeriodicAllowance",
        value: PeriodicAllowance.encode(PeriodicAllowance.fromPartial((content.value as DeepPartial<PeriodicAllowance>))).finish()
      });

    case "cosmos-sdk/AllowedMsgAllowance":
      return Any.fromPartial({
        typeUrl: "/cosmos.feegrant.v1beta1.AllowedMsgAllowance",
        value: AllowedMsgAllowance.encode(AllowedMsgAllowance.fromPartial((content.value as DeepPartial<AllowedMsgAllowance>))).finish()
      });

    default:
      return Any.fromAmino(content);
  }
};
export const FeeAllowanceI_ToAmino = (content: Any) => {
  switch (content.typeUrl) {
    case "/cosmos.feegrant.v1beta1.BasicAllowance":
      return {
        type: "cosmos-sdk/BasicAllowance",
        value: BasicAllowance.toAmino(BasicAllowance.decode(content.value))
      };

    case "/cosmos.feegrant.v1beta1.PeriodicAllowance":
      return {
        type: "cosmos-sdk/PeriodicAllowance",
        value: PeriodicAllowance.toAmino(PeriodicAllowance.decode(content.value))
      };

    case "/cosmos.feegrant.v1beta1.AllowedMsgAllowance":
      return {
        type: "cosmos-sdk/AllowedMsgAllowance",
        value: AllowedMsgAllowance.toAmino(AllowedMsgAllowance.decode(content.value))
      };

    default:
      return Any.toAmino(content);
  }
};