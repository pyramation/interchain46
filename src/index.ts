export * from './codegen';

import { cosmos } from './codegen';

const {
    createGroupWithPolicy
} = cosmos.group.v1.MessageComposer.withTypeUrl;

const msg = createGroupWithPolicy({
    admin: 'a string',
    groupMetadata: 'string here',
    groupPolicyAsAdmin: true,
    groupPolicyMetadata: 'a string here too',
    members: [], // MemberRequest[],
    decisionPolicy: {
        typeUrl: '/cosmos.group.v1.PercentageDecisionPolicy',
        value: {

        }
    }

})