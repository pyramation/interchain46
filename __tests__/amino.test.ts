import { cosmos } from '../src';
import Long from 'long';
import { MsgCreateGroupWithPolicy, MsgCreateGroupWithPolicyEncoded, MsgCreateGroupWithPolicyProtoMsg, DecisionPolicy_FromAmino } from '../src/codegen/cosmos/group/v1/tx';
import { DecisionPolicyWindows, DecisionPolicy_ToAmino, PercentageDecisionPolicyAmino } from '../src/codegen/cosmos/group/v1/types';
import { Msg } from '../src/codegen/cosmos/authz/v1beta1/tx.rpc.msg';

const { createGroupPolicy, createGroupWithPolicy } =
    cosmos.group.v1.MessageComposer.withTypeUrl;
const { PercentageDecisionPolicy } = cosmos.group.v1;

const msg1 = createGroupWithPolicy({
    admin: 'address',
    groupMetadata: 'https://some.ipfs.w3s.link/group_metadata.json',
    groupPolicyAsAdmin: true,
    groupPolicyMetadata: 'https://some.ipfs.w3s.link/group_metadata.json',
    members: [
        {
            address: 'address',
            weight: '10',
            metadata: 'First user'
        }
    ],
    decisionPolicy: PercentageDecisionPolicy.toProtoMsg({
        percentage: '0.5',
        windows: {
            votingPeriod: {
                nanos: 0,
                seconds: Long.fromValue(100)
            },
            minExecutionPeriod: {
                nanos: 0,
                seconds: Long.fromValue(100)
            }
        }
    })
});


const policy: MsgCreateGroupWithPolicyEncoded = {
    admin: 'address',
    groupMetadata: 'https://some.ipfs.w3s.link/group_metadata.json',
    groupPolicyAsAdmin: true,
    groupPolicyMetadata: 'https://some.ipfs.w3s.link/group_metadata.json',
    members: [
        {
            address: 'address',
            weight: '10',
            metadata: 'First user'
        }
    ],
    decisionPolicy: PercentageDecisionPolicy.toProtoMsg({
        percentage: '0.5',
        windows: {
            votingPeriod: {
                nanos: 0,
                seconds: Long.fromValue(100)
            },
            minExecutionPeriod: {
                nanos: 0,
                seconds: Long.fromValue(100)
            }
        }
    })
};

// const policyMsg: MsgCreateGroupWithPolicyProtoMsg = MsgCreateGroupWithPolicy.toProtoMsg(msg)
const protoMsg: MsgCreateGroupWithPolicyProtoMsg = MsgCreateGroupWithPolicy.toProtoMsg(policy)
const msg = {
    typeUrl: '/cosmos.group.v1.MsgCreateGroupWithPolicy',
    value: policy
}



// const msg = createGroupWithPolicy({
//     admin: 'address',
//     groupMetadata: 'https://some.ipfs.w3s.link/group_metadata.json',
//     groupPolicyAsAdmin: true,
//     groupPolicyMetadata: 'https://some.ipfs.w3s.link/group_metadata.json',
//     members: [
//         {
//             address: 'address',
//             weight: '10',
//             metadata: 'First user'
//         }
//     ],
//     decisionPolicy: {
//         typeUrl: '/cosmos.group.v1.PercentageDecisionPolicy',
//         value: PercentageDecisionPolicy.encode({
//             percentage: '0.5',
//             windows: {
//                 votingPeriod: {
//                     nanos: 0,
//                     seconds: Long.fromValue(100)
//                 },
//                 minExecutionPeriod: {
//                     nanos: 0,
//                     seconds: Long.fromValue(100)
//                 }
//             }
//         }).finish()
//     }
// });


it('where it breaks', () => {
    const myObj = MsgCreateGroupWithPolicy.toAmino(msg.value)
    // myObj is correct
    expect(myObj).toMatchSnapshot();

    const encodedVersion = DecisionPolicy_FromAmino(myObj.decision_policy);
    expect({ encodedVersion }).toMatchSnapshot();
    const decodedVersion = DecisionPolicy_ToAmino(encodedVersion);
    expect({ decodedVersion }).toMatchSnapshot();


    const encodedVersion2 = DecisionPolicy_FromAmino(decodedVersion);
    expect({ encodedVersion2 }).toMatchSnapshot();
    const decodedVersion2 = DecisionPolicy_ToAmino(encodedVersion2);
    expect({ decodedVersion2 }).toMatchSnapshot();



});

it('tests', () => {
    const objAmino = MsgCreateGroupWithPolicy.toAmino(msg.value)
    expect(objAmino).toMatchSnapshot();
    expect(objAmino.decision_policy).toMatchSnapshot();

    const encodedVersion = DecisionPolicy_FromAmino(objAmino.decision_policy);
    expect({ encodedVersion }).toMatchSnapshot();

    const decodedVersion = DecisionPolicy_ToAmino(encodedVersion);
    expect({ decodedVersion }).toMatchSnapshot();

    const simple1 = PercentageDecisionPolicy.fromAmino(objAmino.decision_policy.value);
    expect(simple1).toMatchSnapshot();

    const simple2 = PercentageDecisionPolicy.toAmino(simple1);
    expect(simple2).toMatchSnapshot();

    const origAttempt = MsgCreateGroupWithPolicy.fromAmino(objAmino);
    expect(origAttempt).toMatchSnapshot();

    // console.log({ origAttempt })

    const aminoFromOrig = MsgCreateGroupWithPolicy.toAmino(origAttempt);
    // console.log(JSON.stringify({ aminoFromOrig }, null, 2))
    // console.log(JSON.stringify({ objAmino }, null, 2))
    expect(aminoFromOrig).toEqual(objAmino);
});
