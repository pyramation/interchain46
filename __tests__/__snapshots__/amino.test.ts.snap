// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests 1`] = `
{
  "admin": "address",
  "decision_policy": {
    "type": "cosmos-sdk/PercentageDecisionPolicy",
    "value": {
      "percentage": "0.5",
      "windows": {
        "min_execution_period": "100000000000",
        "voting_period": "100000000000",
      },
    },
  },
  "group_metadata": "https://some.ipfs.w3s.link/group_metadata.json",
  "group_policy_as_admin": true,
  "group_policy_metadata": "https://some.ipfs.w3s.link/group_metadata.json",
  "members": [
    {
      "address": "address",
      "metadata": "First user",
      "weight": "10",
    },
  ],
}
`;

exports[`tests 2`] = `
{
  "type": "cosmos-sdk/PercentageDecisionPolicy",
  "value": {
    "percentage": "0.5",
    "windows": {
      "min_execution_period": "100000000000",
      "voting_period": "100000000000",
    },
  },
}
`;

exports[`tests 3`] = `
{
  "encodedVersion": {
    "$typeUrl": "/google.protobuf.Any",
    "typeUrl": "/cosmos.group.v1.PercentageDecisionPolicy",
    "value": {
      "data": [
        10,
        3,
        48,
        46,
        53,
        18,
        8,
        10,
        2,
        8,
        100,
        18,
        2,
        8,
        100,
      ],
      "type": "Buffer",
    },
  },
}
`;

exports[`tests 4`] = `
{
  "decodedVersion": {
    "type": "cosmos-sdk/PercentageDecisionPolicy",
    "value": {
      "percentage": "0.5",
      "windows": {
        "min_execution_period": "100000000000",
        "voting_period": "100000000000",
      },
    },
  },
}
`;

exports[`tests 5`] = `
{
  "percentage": "0.5",
  "windows": {
    "minExecutionPeriod": {
      "nanos": 0,
      "seconds": Long {
        "high": 0,
        "low": 100,
        "unsigned": false,
      },
    },
    "votingPeriod": {
      "nanos": 0,
      "seconds": Long {
        "high": 0,
        "low": 100,
        "unsigned": false,
      },
    },
  },
}
`;

exports[`tests 6`] = `
{
  "percentage": "0.5",
  "windows": {
    "min_execution_period": "100000000000",
    "voting_period": "100000000000",
  },
}
`;

exports[`tests 7`] = `
{
  "admin": "address",
  "decisionPolicy": {
    "$typeUrl": "/google.protobuf.Any",
    "typeUrl": "/cosmos.group.v1.PercentageDecisionPolicy",
    "value": {
      "data": [
        10,
        3,
        48,
        46,
        53,
        18,
        8,
        10,
        2,
        8,
        100,
        18,
        2,
        8,
        100,
      ],
      "type": "Buffer",
    },
  },
  "groupMetadata": "https://some.ipfs.w3s.link/group_metadata.json",
  "groupPolicyAsAdmin": true,
  "groupPolicyMetadata": "https://some.ipfs.w3s.link/group_metadata.json",
  "members": [
    {
      "address": "address",
      "metadata": "First user",
      "weight": "10",
    },
  ],
}
`;

exports[`where it breaks 1`] = `
{
  "admin": "address",
  "decision_policy": {
    "type": "cosmos-sdk/PercentageDecisionPolicy",
    "value": {
      "percentage": "0.5",
      "windows": {
        "min_execution_period": "100000000000",
        "voting_period": "100000000000",
      },
    },
  },
  "group_metadata": "https://some.ipfs.w3s.link/group_metadata.json",
  "group_policy_as_admin": true,
  "group_policy_metadata": "https://some.ipfs.w3s.link/group_metadata.json",
  "members": [
    {
      "address": "address",
      "metadata": "First user",
      "weight": "10",
    },
  ],
}
`;

exports[`where it breaks 2`] = `
{
  "encodedVersion": {
    "$typeUrl": "/google.protobuf.Any",
    "typeUrl": "/cosmos.group.v1.PercentageDecisionPolicy",
    "value": {
      "data": [
        10,
        3,
        48,
        46,
        53,
        18,
        8,
        10,
        2,
        8,
        100,
        18,
        2,
        8,
        100,
      ],
      "type": "Buffer",
    },
  },
}
`;

exports[`where it breaks 3`] = `
{
  "decodedVersion": {
    "type": "cosmos-sdk/PercentageDecisionPolicy",
    "value": {
      "percentage": "0.5",
      "windows": {
        "min_execution_period": "100000000000",
        "voting_period": "100000000000",
      },
    },
  },
}
`;

exports[`where it breaks 4`] = `
{
  "encodedVersion2": {
    "$typeUrl": "/google.protobuf.Any",
    "typeUrl": "/cosmos.group.v1.PercentageDecisionPolicy",
    "value": {
      "data": [
        10,
        3,
        48,
        46,
        53,
        18,
        8,
        10,
        2,
        8,
        100,
        18,
        2,
        8,
        100,
      ],
      "type": "Buffer",
    },
  },
}
`;

exports[`where it breaks 5`] = `
{
  "decodedVersion2": {
    "type": "cosmos-sdk/PercentageDecisionPolicy",
    "value": {
      "percentage": "0.5",
      "windows": {
        "min_execution_period": "100000000000",
        "voting_period": "100000000000",
      },
    },
  },
}
`;
